### PR TITLE
Add validation for assignability of Object types to Ballerina types when using runtime APIs

### LINKED ISSUE #34195: `java.lang.ClassCastException` when incorrectly return a distinct error in extern function
**Description:**
$title 
CCE should be handled properly.

**Steps to reproduce:**
Java code:
```
public class App {

    private static Module errorModule = new Module("org", "interop_project.errors", "1");

    public static BError getDistinctErrorUnion(BString errorName) {
        BMap<BString, Object> errorDetails = ValueCreator.createMapValue();
        errorDetails.put(StringUtils.fromString("detail"), "detail error");
        return ErrorCreator.createError(errorModule, errorName.getValue(), StringUtils.fromString("msg"),
                null, errorDetails);
    }
}
```
Ballerina code:
errors.bal
```
import ballerina/jballerina.java;

public type DistinctError distinct error;

public type Error distinct error;

public function getDistinctErrorUnion(string msg) returns Error = @java:Method {
    'class: "javalibs.app.App"
} external;
```
main.bal
```
import interop_project.errors;

public function main() {
    errors:Error err = errors:getDistinctErrorUnion("DistinctError");
}
```

error:
```
error: java.lang.ClassCastException {"message":"class java.lang.String cannot be cast to class io.ballerina.runtime.api.values.BValue (java.lang.String is in module java.base of loader 'bootstrap'; io.ballerina.runtime.api.values.BValue is in unnamed module of loader 'app')"}
        at org.interop_project.errors.1:getDistinctErrorUnion(errors.bal:7)
           org.interop_project.1:main(main.bal:4)
```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### LINKED ISSUE #34587: CCE with Java interop
**Description:**
CCE with Java interop when java.util.ArrayList is provided as map entry to create a record value.

**Steps to reproduce:**
Java code:
```
public class App {

    private static Module module = new Module("org", "interop_project.records", "1");

    public static BMap<BString, Object> getRecord(BString recordName) {
        ArrayList<Integer> arrayList = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
        Map<String, Object> map = Map.ofEntries(
                Map.entry("arrList", arrayList)
        );
        return ValueCreator.createRecordValue(module, recordName.getValue(), map);
    }
}
```
Ballerina code:
module/records.bal
```
import ballerina/jballerina.java;

public type Foo record {
    int[] x;    
};

public function getRecord(string recordName) returns record{} = @java:Method {
    'class: "javalibs.app.App"
} external;
```
main.bal
```
import interop_project.records;

public function main() {
    records:Foo foo =  <records:Foo> records:getRecord("Foo");
}
```

error:
```
error: java.lang.ClassCastException {"message":"class java.util.ArrayList cannot be cast to class io.ballerina.runtime.api.values.BValue (java.util.ArrayList is in module java.base of loader 'bootstrap'; io.ballerina.runtime.api.values.BValue is in unnamed module of loader 'app')"}
        at org.interop_project.records.1:getRecord(records.bal:7)
           org.interop_project.1:main(main.bal:4)
```

**Affected Versions:**
slbeta3, slbeta4, slbeta6 

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #34587
Fixes #34195

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java [hunk 1]
```diff
@@ -197,6 +201,15 @@ public static String getMajorVersion(String version) {
         return version.split("\\.")[0];
     }
 
+    public static void validateObjectAssignabilityToBType(BMap<BString, Object> bMap) {
+        for (Object value : bMap.values()) {
+            if (value != null && !(value instanceof Number) && !(value instanceof BString) &&
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java:206]: We can have a method in type checker with isValidBallerinaValue and use in createRecordvalue and here

File: bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties [hunk 2]
```diff
@@ -183,6 +183,7 @@ object.invalid.final.field.update = cannot update ''final'' field ''{0}'' in obj
 unsupported.comparison.operation = cannot compare ''{0}'' and ''{1}''
 unordered.types.in.comparison = ''{0}'' is unordered with respect to ''{1}''
 incompatible.type.assignment = ''{0}'' cannot be assigned to type ''{1}''
+incompatible.java.object.type.assignment = value creation with ''{0}'' not supported: expected a BValue
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties:186]: ''{0}'' is not from a valid java runtime class. It should be sub class of one of following classes.

java.lang.Number
java.lang.Float 
or
package io.runtime.api.values.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java [hunk 3]
```diff
@@ -197,6 +202,28 @@ public static String getMajorVersion(String version) {
...
+        for (Map.Entry<String, Object> fieldEntry : bMap.entrySet()) {
+            Object value = fieldEntry.getValue();
+            if (isInvalidBallerinaValue(value) && !(value instanceof String)) {
+                throw ErrorUtils.createJToBTypeCastError(value.getClass());
+            }
+        }
+    }
+
+    private static boolean isInvalidBallerinaValue(Object value) {
+        return (value != null && !(value instanceof Number) && !(value instanceof BString) &&
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java:223]: we can remove (value instanceof BString)

File: bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties [hunk 4]
```diff
@@ -183,6 +183,8 @@ object.invalid.final.field.update = cannot update ''final'' field ''{0}'' in obj
 unsupported.comparison.operation = cannot compare ''{0}'' and ''{1}''
 unordered.types.in.comparison = ''{0}'' is unordered with respect to ''{1}''
 incompatible.type.assignment = ''{0}'' cannot be assigned to type ''{1}''
+incompatible.java.object.type.assignment = ''{0}'' is not from a valid java runtime class. It should be subclass of \
+  one of the following: java.lang.Number, java.lang.Boolean or package io.runtime.api.values
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties:187]:
```suggestion
incompatible.java.object.type.assignment = ''{0}'' is not from a valid java runtime class. It should be a subclass of \
 one of the following: java.lang.Number, java.lang.Boolean or from the package 'io.runtime.api.values'
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java [hunk 5]
```diff
@@ -76,6 +76,7 @@
         BMap<BString, Object> record = createRecordValue(packageId, recordTypeName);
         for (Map.Entry<String, Object> fieldEntry : valueMap.entrySet()) {
             Object val = fieldEntry.getValue();
+            // TODO: Remove the following String to BString conversion.
```
[the remark this answers is not in the document]
Comment (gabilang) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java:79]: Created a separate [task](https://github.com/ballerina-platform/ballerina-lang/issues/34882) for that.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/exceptions/RuntimeErrors.java [hunk 6]
```diff
@@ -122,7 +122,8 @@
     CONFIG_CLI_ARGS_AMBIGUITY("config.cli.args.ambiguity", "RUNTIME_0094"),
     CONFIG_CLI_UNUSED_CLI_ARGS("config.cli.unused.args", "RUNTIME_0095"),
     LARGE_EXPONENTS_IN_DECIMAL("large.number.of.exponents.in.decimal", "RUNTIME_0096"),
-    UNSUPPORTED_DECIMAL_VALUE("unsupported.decimal.value", "RUNTIME_0097");
+    UNSUPPORTED_DECIMAL_VALUE("unsupported.decimal.value", "RUNTIME_0097"),
+    J_TYPE_ASSIGNABLE_ERROR("incompatible.java.object.type.assignment", "RUNTIME_0098");
```
Comment (Nadeeshan96) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/exceptions/RuntimeErrors.java:126]: There is a space above in the latest runtime-fixes branch which is for the code RUNTIME_0041. Can you move this there

**Pull request merged**